src/components/error-overlay/error-message-id-provider.ts

```diff
@@ -0,0 +1,33 @@
+type ErrorDetails = {
```

Comment:
i think you can rename the file to get-error-message-by-code (get-error-message-id-by-category) just so it would be clear from the filename what it contains, and put it in /utils, so the logic can be reused later